### PR TITLE
feat: support passing in access tokens instead of client_id and client_secret

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

File: redpanda/redpanda.go [hunk 1]
```diff
@@ -117,11 +131,22 @@ type credentials struct {
...
+	// - If the provider configuration has a client id/secret pair, but a token is provided in an environment
+	//   variable, should we use the token if it validates against the client id?
+	// - Should we keep the client id/secret around as well in case we need to re-auth with the API?
+	// - If the user passes in a token, should we validate its audience field against the configured
+	//   Redpanda cloud environment/endpoint? or, should we pull the endpoint from the audience field? or
+	//   do nothing like we do now?
+
 	endpoint, err := cloud.EndpointForEnv(cloudEnv)
 	if err != nil {
 		diags.AddError(err.Error(), "")
```
Comment (gene-redpanda) [inline redpanda/redpanda.go:152]: We really should have a searchable error summary message here

**Comment by gene-redpanda** [comment]
This overall looks pretty good to me, just needs to pass lint.

File: redpanda/redpanda.go (hunk 1 above)
Comment (ligfx) [reply redpanda/redpanda.go:152]: For the token vs. environment bit specifically?

**Comment by ligfx** [comment]
Not sure what's up with the `make test_network` failure, it works fine for me locally when I use my own integration env creds.

File: redpanda/redpanda.go [hunk 2]
```diff
@@ -76,22 +82,134 @@ func New(_ context.Context, cloudEnv, version string) func() provider.Provider {
...
+	// - If the user passes in a token, should we validate its audience field against the configured
+	//   Redpanda cloud environment/endpoint? or, should we pull the endpoint from the audience field? or
+	//   do nothing like we do now?
+
+	creds := credentials{}
+	diags := diag.Diagnostics{}
+
+	endpoint, err := cloud.EndpointForEnv(cloudEnv)
+	if err != nil {
+		diags.AddError(err.Error(), "")
```
Comment (gene-redpanda) [inline redpanda/redpanda.go:155]:
```suggestion
		diags.AddError("error retrieving correct endpoint, err.Error())
```

**Comment by gene-redpanda** [comment]
> Not sure what's up with the `make test_network` failure, it works fine for me locally when I use my own integration env creds.

You are missing the preprod testing changes which means the CI will fail here.

File: redpanda/redpanda.go (hunk 2 above)
Comment (ligfx) [reply redpanda/redpanda.go]: Done

**Review by gene-redpanda — APPROVED** [review]
LGTM

**Pull request merged**